lib/Tools/LinearLayout.cpp

```diff
@@ -918,6 +927,13 @@ LinearLayout LinearLayout::invertAndCompose(const LinearLayout &outer) const {
   return flatComposed.reshapeIns(retInDims).reshapeOuts(retOutDims);
 }
 
+LinearLayout LinearLayout::invert() const {
+  SmallVector<std::pair<StringAttr, int32_t>> dimsAndSizes;
+  llvm::append_range(dimsAndSizes, outDims);
+  auto id = identityND(dimsAndSizes);
+  return id.invertAndCompose(*this);
+}
```

Comment:
Not sure if this will work as expected before https://github.com/triton-lang/triton/pull/5309.
I will try to have it merged today tho